### PR TITLE
document flavors support for macos

### PR DESCRIPTION
**Pull request by andrewkolos**
Fixes https://github.com/flutter/website/issues/9988.

The setup process for flavors of macOS apps is nearly-identical to that of iOS apps, so we can update the existing iOS sections to also apply to macOS.

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/deployment/flavors.md [hunk 1]
```diff
@@ -40,9 +40,9 @@ Prerequisites:
 * Xcode installed
 * An existing Flutter project
 
-To set up flavors in iOS, you'll define build configurations in Xcode.
+To set up flavors in iOS or macOS, you'll define build configurations in Xcode.
 
-## Creating flavors in iOS
+## Creating flavors in iOS or macOS
```
Comment (andrewkolos) [inline src/deployment/flavors.md:45]: Is this an appropriate use of "or"?

File: src/deployment/flavors.md [hunk 2]
```diff
@@ -143,9 +143,10 @@ section at the end of this document.
 
 ### Plugin configurations
 
-If your app uses a Flutter plugin, you need to update the `ios/Podfile`.
+If your app uses a Flutter plugin, you need to update
+`ios/Podfile` (if developing for iOS) and `macos/Podfile` (if developing for macOS).
```
Comment (andrewkolos) [inline src/deployment/flavors.md:147]: Should "(if developing for iOS)" be shortened to "(for iOS)"?

File: src/deployment/flavors.md [hunk 3]
```diff
@@ -143,9 +143,10 @@ section at the end of this document.
...
 
-1. In `ios/Podfile` change the default for
+1. In `ios/Podfile`/`macos/Podfile` change the default for
```
Comment (andrewkolos) [inline src/deployment/flavors.md:149]: Note for myself/reviewer for when page preview is generated: is this slash readable?

Comment (andrewkolos) [reply src/deployment/flavors.md]: This wasn't very readable, so I replaced it with an "and".

File: src/deployment/flavors.md (hunk 1 above)
Comment (domesticmouse) [reply src/deployment/flavors.md:45]: I'd go with and.

**Review by domesticmouse — APPROVED** [review]
PTAL @sfshaza2

**Review by atsansone — APPROVED** [review]
Thanks for these updates! Any other issues we'll handle with the other Flavor issues to be resolved.

**Pull request merged**